File: app/src/main/java/com/githang/gradledoc/app/contents/ChapterUrlModel.java
package com.githang.gradledoc.app.contents;

import com.githang.gradledoc.common.Model;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

import java.util.ArrayList;
import java.util.List;

/**
 * @author 黄浩杭 (huanghaohang@parkingwang.com)
 * @version 2017-04-08
 * @since 2017-04-08
 */

class ChapterUrlModel extends Model<List<ChapterUrl>> {
    @Override
    public List<ChapterUrl> handleContent(String content) {
        Document doc = Jsoup.parse(content);
        Elements elements = doc.select("span.chapter");
        List<ChapterUrl> chapterUrls = new ArrayList<>();
        for (Element elem : elements) {
            ChapterUrl url = new ChapterUrl();
            url.setUrl(elem.select("a[href]").attr("href"));
            url.setTitle(elem.text());
            chapterUrls.add(url);
        }
        return chapterUrls;
    }
}

File: app/src/main/java/com/githang/gradledoc/app/contents/ChapterUrlModel.kt
package com.githang.gradledoc.app.contents

import com.githang.gradledoc.common.Model
import org.jsoup.Jsoup
import java.util.ArrayList

/**
 * @author 黄浩杭 (huanghaohang@parkingwang.com)
 * @version 2017-04-08
 * @since 2017-04-08
 */
internal class ChapterUrlModel : Model<List<ChapterUrl>>() {
    override fun handleContent(content: String): List<ChapterUrl> {
        val doc = Jsoup.parse(content)
        val elements = doc.select("span.chapter")
        val chapterUrls = ArrayList<ChapterUrl>()
        for (elem in elements) {
            val url = ChapterUrl(elem.text(), elem.select("a[href]").attr("href"))
            chapterUrls.add(url)
        }
        return chapterUrls
    }
}
